src/lib.rs

```diff
@@ -501,6 +616,102 @@ where
     }
 }
 
+impl<T, const BITS: usize> Mul for UInt<T, BITS>
+where
+    Self: Number,
+    T: PartialEq
+        + Copy
+        + BitAnd<T, Output = T>
+        + Not<Output = T>
+        + Mul<T, Output = T>
+        + Sub<T, Output = T>
+        + Shr<usize, Output = T>
+        + Shl<usize, Output = T>
+        + From<u8>,
+{
+    type Output = UInt<T, BITS>;
+
+    fn mul(self, rhs: Self) -> Self::Output {
+        let product = self.value * rhs.value;
```

Comment:
This has another overflow check in debug, right? We need it, but should have add a comment that there is another check that we can’t avoid?